blenderproc/python/loader/HavenMaterialLoader.py

```diff
@@ -21,65 +100,51 @@ def load_haven_mat(folder_path: str = "resources/haven", used_assets: list = [],
-                        base_image_path = path
-                        break
-                if not os.path.exists(base_image_path):
-                    continue
-
-                # if the material was already created it only has to be searched
-                if fill_used_empty_materials:
-                    new_mat = MaterialLoaderUtility.find_cc_material_by_name(asset, add_cp)
-                else:
-                    new_mat = MaterialLoaderUtility.create_new_cc_material(asset, add_cp)
-                if preload:
-                    # if preload then the material is only created but not filled
-                    continue
-                elif fill_used_empty_materials and not MaterialLoaderUtility.is_material_used(new_mat):
-                    # now only the materials, which have been used should be filled
-                    continue
-
-                # construct all image paths
-                # the images path contain the words named in this list, but some of them are differently
-                # capitalized, e.g. Nor, NOR, NoR, ...
-                used_elements = ["ao", "spec", "rough", "nor", "disp", "bump", "alpha"]
-                final_paths = {}
-                for ele in used_elements:
-                    new_path = base_image_path.replace("diff", ele).lower()
-                    found_path = ""
-                    for path in all_paths:
-                        if path.lower() == new_path:
-                            found_path = path
-                            break
-                    final_paths[ele] = found_path
-
-                # create material based on these image paths
-                HavenMaterialLoader.create_material(new_mat, base_image_path, final_paths["ao"],
-                                                    final_paths["spec"], final_paths["rough"],
-                                                    final_paths["alpha"], final_paths["nor"],
-                                                    final_paths["disp"], final_paths["bump"])
-    else:
-        raise Exception("The folder path does not exist: {}".format(folder_path))
+        
+    if not os.path.exists(textures_folder_path) or not os.path.isdir(textures_folder_path):
+        raise Exception("The folder path does not exist: {}".format(textures_folder_path))
+
+    texture_names = os.listdir(folder_path)
+    for texture_name in texture_names:       
+        if used_assets and not any(texture_name.startswith(asset) for asset in used_assets):
+            continue
+
+        texture_folder_path = os.path.join(textures_folder_path, texture_name)
+        if not os.path.isdir(texture_folder_path):
+            print(f"Ignoring {texture_folder_path}, must be a folder.")
+            continue
+
+        texture_map_paths_by_type = identify_texture_maps(texture_folder_path)
+        if texture_map_paths_by_type is None:
+            print(f"Ignoring {texture_name}, could not identify texture maps.")
```

Comment:
This currently only happens when no base color image could be found in the `texture_folder_path` with `"diff"`, `"diffuse"`, `"col"` or `"albedo"`. This could happen for several reasons e.g. the naming convention is different, the download failed, the user deleted the file etc.

Reply:
This could also happens when a user places all their textures in the same folder. Then non-Haven textures would be mixed with Haven textures and we would fail to identify all the texture maps.

Reply:
Makes sense